hooks/_common.sh

```diff
@@ -170,6 +170,59 @@ function common::is_hook_run_on_whole_repo {
   fi
 }
 
+#######################################################################
+# Get the number of CPU logical cores available to pre-commit use
+# Arguments:
+#  parallelism_ci_cpu_cores (string) Used in corner cases when CPU cores
+#    can't be calculated
+# Outputs:
+#   Return number CPU logical cores, rounded to below integer
+#######################################################################
+function common::get_cpu_num {
+  local -r parallelism_ci_cpu_cores=$1
+
+  local millicpu
+
+  if [[ -f /sys/fs/cgroup/cpu/cpu.cfs_quota_us ]]; then
+    # Inside K8s pod or DInD in K8s
+    millicpu=$(cat /sys/fs/cgroup/cpu/cpu.cfs_quota_us)
+
+    if [[ $millicpu -eq -1 ]]; then
+      # K8s no limits or in DinD
+      if [[ ! $parallelism_ci_cpu_cores ]]; then
+        common::colorify "yellow" "Unable to calculate available CPU cors.\n" \
+          "You in K8s pod without limits or in DinD without limits propagation.\n" \
+          "To avoid possible harm, parallelism disabled.\n" \
+          "If you'd like reenable it - set corresponding limits, or set next for current hook:\n" \
```

Comment:
It's a bad manners to have a one-way conversations with users from within scripts =)
```suggestion
          "To re-enable it, set corresponding limits, or set the following for the current hook:\n" \
```